rules/no-await-in-loop.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview Rule to disallow uses of await inside of loops.
+ * @author Nat Mote
+ */
+"use strict";
+
+// Node types which are considered loops.
+var loopTypes = new Set([
```

Comment:
there is nothing set in stone but I think we are mostly in keeping with eslint, which still supports 0.10.x i believe...my rule of thumb has been: es5